src/ceph_fuse.cc

```diff
@@ -115,28 +115,28 @@ int main(int argc, const char **argv, const char *envp[]) {
     cerr << std::endl;
   }
 
-  global_init_prefork(g_ceph_context);
   Preforker forker;
   if (g_conf->daemonize) {
-    string err;
-    if (forker.prefork(err)) {
-      cerr << "ceph-fuse[" << err << std::endl;
-      return 1;
-    }
-    global_init_postfork_start(cct.get());
-  }
-
-
-  if (forker.is_parent()) {
-    string err;
-    int r = forker.parent_wait(err);
-    if (r) {
-      cerr << "ceph-fuse" << err << std::endl;
+    if (global_init_prefork(g_ceph_context) >= 0) {
+      int r;
+      string err;
+      r = forker.prefork(err);
+      if (r < 0) {
+	cerr << "ceph-fuse" << err << std::endl;
+	forker.exit(r);
```

Comment:
the exit(3) call does not allow the destructors to be called. take the `cct` for example, its dtor won't be called this way. so i'd recommend use `return` instead.